include/glow/Importer/CommonOperatorLoader.h

```diff
@@ -475,58 +475,59 @@ class CommonOperatorLoader : public ProtobufLoader {
     bool broadcast;
     ASSIGN_VALUE_OR_RETURN_ERR(broadcast, getBroadcast(dict));
 
-    NodeValue finalIn0 = in0;
-    NodeValue finalIn1 = in1;
+    int axis = -1;
 
     if (broadcast) {
       // Broadcasting can be:
       // - multidirectional (ONNX opset 7+), or
       // - unidirectional (ONNX opset 1->6,  Caffe2).
-      if (hasMultidirectionalBroadcast(typeName)) {
-        // Compute the target shape that is a combination of the operand shapes.
-        std::vector<size_t> targetDim;
-        ASSIGN_VALUE_OR_RETURN_ERR(targetDim, computeMultidirectionalBroadcast(
-                                                  in0.dims(), in1.dims()));
-        // Sets the axis of each inputs so that the trailing-most dimensions of
-        // input tensors and the target shape are aligned.
-        int axis0 = targetDim.size() - in0.dims().size();
-        int axis1 = targetDim.size() - in1.dims().size();
-        finalIn0 = G_.createBroadcast(opName, in0, targetDim, axis0);
-        finalIn1 = G_.createBroadcast(opName, in1, targetDim, axis1);
-      }
+
       // Unidirectional broadcasting consists of broadcasting the right operand
       // (in1) so that it matches the shape of the left operand (in0).
-      else {
+      if (!hasMultidirectionalBroadcast(typeName)) {
         // With unidirectional broadcasting, the 'axis' attribute specifies
         // from how much the right operand shape must be 'shifted' right.
         // - In Caffe2, the 'axis' attribute is optional. If not specified, axis
         // must be automatically computed so that the trailing-most dimensions
         // of in1 is aligned to the trailing-most dimension of in0.
         // - In ONNX, the 'axis' attribute is mandatory. axis == -1 is
         // equivalent to no axis specified in Caffe2.
-        int axis = -1;
+
         if (dict.count("axis")) {
           ASSIGN_VALUE_OR_RETURN_ERR(axis, loadInt(dict["axis"]));
         }
         if (axis == -1) {
           // Align trailing most dimensions.
           axis = in0.dims().size() - in1.dims().size();
         }
-        finalIn1 = G_.createBroadcast(opName, in1, in0.dims(), axis);
       }
     }
 
     Node *node = nullptr;
-    if (typeName == "Mul") {
-      node = G_.createMul(opName, finalIn0, finalIn1);
-    } else if (typeName == "Add") {
-      node = G_.createAdd(opName, finalIn0, finalIn1);
-    } else if (typeName == "Sub") {
-      node = G_.createSub(opName, finalIn0, finalIn1);
-    } else if (typeName == "Div") {
-      node = G_.createDiv(opName, finalIn0, finalIn1);
+    if (broadcast) {
```

Comment:
If we're broadcasting, why can't we just call `broadcastInputs(axis, {in0, in1});` and then set `in0` and `in1` from the returned vector? Then we don't need all of the complicated macro/template stuff from `Graph.h`, and we can also call `node = G_.createMul(opName, in0, in1);` etc. without special casing it from broadcast vs not. E.g.:

```cpp
if (broadcast) {
  auto broadcastedInputs = broadcastInputs(axis, {in0, in1});
  assert(broadcastedInputs.size() == 2 && "...");
  in0 = broadcastedInputs[0];
  in1 = broadcastedInputs[1];
}
if (typeName == "Mul") {
  node = G_.createMul(opName, in0, in1);
} else if (typeName == "Add") {
  node = G_.createAdd(opName, in0, in1);
} else if (typeName == "Sub") {
  node = G_.createSub(opName, in0, in1);
} else if (typeName == "Div") {
  node = G_.createDiv(opName, in0, in1);
} else {
  RETURN_ERR("Unsupported arithmetic typeName");
}
```

Reply:
We then have to replicate this logic here in OperatorTests, in Less, and Where. VS this approach it's all encapsulated in one place "under the hood" in one templated method.